docs/config-docs.md

```diff
@@ -0,0 +1,374 @@
+```xml
+<box>
+	{{potatoes}}
+</box>
+```
+You can also change the variable name in CLI by doing:
+```sh
+eww update potatoes carrots
+```
+This will change the variable name `potatoes` to `carrots`
+
+#### The `<script-var>` tag 
+
+Allows you to create a script that eww runs.
+Useful for creating volume sliders or anything similar.
+
+Example: 
+```xml
+<variables>
+	<script-var name="date" interval="5s">
+	date +%H:%M
+	</script-var>
+</variables>
+```
+
+and then reference it by doing:
+```xml
+<box>
+	{{date}}
+</box>
+```
+
+The `interval="5s"` part says how long time it should take before Eww runs the command again.
+Here are the available times you can set:
+| Shortened | Full name | 
+| :------------- | :----------:|  
+| ms | Miliseconds |  
+| s | Seconds |
+| m | Minutes |  
+| h | Hours | 
+
+#### Tail
+If you don't want a set interval and instead want it to tail (run the script when it detects a change is present) you can simply remove the `interval="5s"` so it becomes:
+```xml
+<variables>
+	<script-var name="date">
+	date +%H:%M
+	</script-var>
+</variables>
+```
+
+### The `<definitions>` block
+In here you whole widget will be made, and you can also create your own widgets. Check [Widget Documentation]() for pre-defined widgets.
+
+#### Custom widgets
+
+Let's get a small config and break it down.
+
+```xml
+<definitions>
+	<def name="clock">
+		<box>
+			The time is: {{my_time}} currently.
+		</box>
+	</def>
+	<def name="main">
+		<box>
+			<clock my_time="{{date}}"/>
+		</box>
+	</def>
+</definitions>
+
+<variables>
+	<script-var name="date">
+		date
+	</script-var>
+</variables>
+```
+That's a long config just for a custom widget. But let's break it down and try to understand it.
+
+This part:
+```xml
+	<def name="clock">
+		<box>
+			The time is: {{my_time}} currently.
+		</box>
+	</def>
+```
+Is the custom widget. As we can see by the 
+```xml 
+<def name="clock">
+```
+the widget is called `clock.`Or referenced `<clock>`
+The `{{my_time}}` is the value we assign to be well, our time. You can actually set to be anything, it doesn't have to be a time. You can compare it to `value=""`
+
+So if we look at:
+```xml
+	<def name="main">
+		<box>
+			<clock my_time="{{date}}"/>
+		</box>
+	</def>
+```
+we can see that we assign `{{my_time}}` to be `{{date}}` and if we look at 
+```xml
+	<script-var name="date">
+		date
+	</script-var>
+```
+we can see that `{{date}}` is simply running the `date` command.
+
+It doesn't have to be `{{my_time}}` either, it can be anything.
+```xml
+	<def name="clock">
+		<box>
+			The time is: {{very_long_list_of_animals}} currently.
+		</box>
+	</def>
+```
+is valid.
+
+To use that it would look like this:
+```xml
+	<def name="main">
+		<box>
+			<clock very_long_list_of_animals="{{date}}"/>
+		</box>
+	</def>
+```
+### The `<windows>` block
+
+This is the part the Eww reads and loads. The `<windows>` config should look something like this:
+
+```xml
+<windows>
+      <window name="main_window">
```

Comment:
```suggestion
      <window name="main_window" stacking="fg">
```
This should mention the stacking option

Reply:
This should also be included in the explaination of the different options later on.
Possible values here are `foreground`, `fg`, `background` and `bg`. These decide, if the window will be shown in front of or behind other windows